docs/plain-bundle-spec.md

```diff
@@ -0,0 +1,101 @@
+# Plain Bundle Spec
+
+## Overview
+
+This document is meant to define the plain bundle format as a reference for those publishing plain bundles for use with
+RukPak. A bundle is a collection of Kubernetes resources that are packaged into a container image. These bundles can be
+unpacked onto a running cluster, where controllers can then create the underlying content embedded in the bundles.
+
+Plain bundles are simply container images containing a set of static, arbitrary, Kubernetes YAML manifests in a given
+directory. The currently implemented plain bundle schema is the `plain+v0` format. The name of the bundle
+format, `plain+v0` combines the type of bundle (plain) with the current schema version (v0).
+The [plain bundle provisioner](/internal/provisioner/plain/README.md) is able to source these `plain+v0` bundles and
+install them onto a Kubernetes cluster.
+
+> Note: the plain+v0 bundle format is at schema version v0, which means it's an experimental format that is subject
+> to change.
+
+## Example
+
+For example, below is a minimal example of a Dockerfile that builds a `plain+v0` bundle from a directory containing
+static Kubernetes manifests.
+
+```dockerfile
+FROM scratch
+COPY /manifests /manifests
+```
+
+where the given `manifests` directory contains the Kubernetes resources required to deploy an application, for example:
+
+```
+manifests
+├── namespace.yaml
+├── cluster_role.yaml
+├── role.yaml
+├── serviceaccount.yaml
+├── cluster_role_binding.yaml
+├── role_binding.yaml
+└── deployment.yaml
+```
+
+## Technical Details
+
+* The static manifests must be located in the root-level /manifests directory in the bundle image for the bundle to be a
+  valid `plain+v0` bundle that the provisioner can unpack. A plain bundle image without a /manifests directory is
+  invalid and will not be successfully unpacked onto the cluster.
+* The /manifests directory may contain additional sub-directories which contain static manifests.
+* The plain bundle can be built from any base image, but `scratch` is recommended as it keeps the resulting bundle image
+  a minimal size.
+* Additional bundle metadata can be provided in two ways: either via image labels or by including a `/metadata`
+  directory inside the bundle. For a`plain+v0` bundle either is acceptable and entirely optional.
+
+Image labels can be used to describe the type of bundle image as well as the bundle name and version. For example, below
+is a set of optional labels which can be applied to the bundle image:
+
+```dockerfile
+LABEL core.rukpak.io.bundle.mediatype.v0=plain+v0
+LABEL core.rukpak.io.bundle.name.v0=my-operator
+LABEL core.rukpak.io.bundle.version.v0=0.0.1
```

Comment:
I hate to be a broken record: can we remove these labels as they're essentially a no-op right now (and there's no concrete evidence, at least for now, that these labels will be needed)?

Reply:
If we're interesting in attaching metadata right now to an individual Bundle, maybe use labels/annotations on the actual Bundle resource?

Reply:
Ok, no problem, I'll take them out -- I think the reason I'm so interested in the labels is because they are used heavily in the `registry+v1` format and I think they may eventually be useful. But as you stated, there's no evidence that they are useful for the plain+v0 format right now and we can drop them for now.

Reply:
Yep - this can always be an additive change if we ever find the need for exposing container-level labels like we do with the registry+v1 format.